core/src/main/java/edu/ucr/cs/riple/core/metadata/index/Error.java

```diff
@@ -33,20 +36,36 @@ public class Error extends Enclosed {
   public final String messageType;
   /** Error message. */
   public final String message;
+  /**
+   * If non-null, this error involved a pseudo-assignment of a @Nullable expression into a @NonNull
+   * target, and this field is the Symbol for that target.
+   */
+  @Nullable private final Location location;
 
   /**
    * NullAway serializes error on TSV file, this constructor is called for each line of that file.
    *
    * @param values Values in row of a TSV file.
    */
   public Error(String[] values) {
-    this(values[0], values[1], values[2], values[3]);
+    this(
+        values[0],
+        values[1],
+        values[2],
+        values[3],
+        Location.createLocationFromArrayInfo(Arrays.copyOfRange(values, 4, 10)));
   }
 
-  public Error(String messageType, String message, String encClass, String encMethod) {
+  public Error(
+      String messageType,
+      String message,
+      String encClass,
+      String encMethod,
+      @Nullable Location nonnullTargetLocation) {
     super(encClass, encMethod);
     this.messageType = messageType;
     this.message = message;
+    this.location = nonnullTargetLocation;
   }
 
   @Override
```

Comment:
Are `equals(...)` and `hashCode(...)` out of date here? They don't even seem to take `message` into account, just `messageType`, so it feels like these would often clash for non-equivalent `Error` objects. (These get used when comparing two `Bank<Error>` collections, so not sure if we are missing stuff there, maybe it is fine if we are only counting the errors of the same type before and after, but then we should use something other than `.equals(...)` since we aren't really checking for equality here)

Reply:
Yes, they are updated thanks for the note, the reason that it was working correctly so far, is that we are only looking at the number of differences in number of errors, and they weren’t actually checked if they are equal. This hash function was there just for `Bank` to be able to compute a hash value for an `Error` instance. 

We weren’t even processing `Message Type` since we determine if an error is resolvable by checking if a new fix is triggered.

Both methods are updated now corresponding to the added fields, both to avoid clashing and in GH-53 we do need to see the difference in errors contents.